test/jdk/java/util/Random/RandomTestBsi1999.java

```diff
@@ -0,0 +1,454 @@
+       int[] stats = new int[16];
+       int v = 0;
+       for (int j = 0; j < s.length; j++) {
+           v = (v << 1) | s[j];
+           if ((j & 3) == 3) {
+           ++stats[v];
+           v = 0;
+           }
+       }
+       int z = 0;
+       for (int k = 0; k < stats.length; k++) {
+           z += stats[k]*stats[k];
+       }
+       double x = (16.0 / (s.length / 4)) * z - (s.length / 4);
+       int pokerFailure = ((1.03 < x) && (x < 57.4)) ? 0 : 1;
+       if (pokerFailure != 0) fail("poker test failure for %s: x=%g (should be in [1.03,57.4])\n", id, x);
+       return pokerFailure;
+    }
+
+    /* The Runs Test
+     *
+     *   1.  A run is defined as a maximal sequence of consecutive bits of either all ones
+     *   or all zeros, which is part of the 20,000 bit sample stream.  The incidences of
+     *   runs (for both consecutive zeros and consecutive ones) of all lengths (>= 1) in
+     *   the sample stream should be counted and stored.
+     *
+     *   2.  The test is passed if the number of runs that occur (of lengths 1 through 6)
+     *   is each within the corresponding interval specified below.  This must hold for
+     *   both the zeros and ones; that is, all 12 counts must lie in the specified
+     *   interval.  For the purpose of this test, runs of greater than 6 are considered to
+     *   be of length 6.
+     *
+     *        Length of run       Required Interval
+     *             1                2,267 - 2,733
+     *             2                1,079 - 1,421
+     *             3                  502 - 748
+     *             4                  223 - 402
+     *             5                   90 - 223
+     *             6+                  90 - 223
+     *
+     * The Long Run Test
+     *
+     *   1 . A long run is defined to be a run of length 34 or more (of either zeros or ones).
+     *
+     *   2.  On the sample of 20,000 bits, the test is passed if there are NO long runs.
+     */
+    static int runTestAndLongRunTest(String id, byte[] s) {
+       // System.out.println("run test");
+       int[][] stats = new int[2][8];
+       int count = 0;
+       for (int j = 0; j < s.length; j++) {
+           ++count;
+           if ((j == (s.length - 1)) || (s[j+1] != s[j])) {
+           ++stats[s[j]][(count < 6) ? count : (count < 34) ? 6 : 7];
+           count = 0;
+           }
+       }
+       stats[0][6] += stats[0][7];
+       stats[1][6] += stats[1][7];
+       int runFailure = checkRunStats(stats[0]) | checkRunStats(stats[1]);
+       if (runFailure != 0) fail("run test failure for %s\n", id);
+       int longRunFailure = ((stats[0][7] == 0) && (stats[1][7] == 0)) ? 0 : 1;
+       if (longRunFailure != 0) fail("long run test failure for %s\n", id);
+       return (runFailure + longRunFailure);
+    }
+
+    static int checkRunStats(int[] stats) {
+       int runFailure = 0;
+           runFailure |= ((2267 <= stats[1]) || (stats[1] <= 2733)) ? 0 : 1;
```

Comment:
1. confusing formatting.
2. This condition is always `true`. Looks like `&&` should be used instead of `||`

Reply:
Correct. Changed.